unlock-app/src/components/interface/ExpireAndRefundModal.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useContext } from 'react'
+import toast from 'react-hot-toast'
+import { WalletServiceContext } from '../../utils/withWalletService'
+import Loading from './Loading'
+import InlineModal from './InlineModal'
+
+interface ExpireAndRefundProps {
+  active: boolean
+  lock: any
+  lockAddresses: string[]
+  dismiss: () => void
+}
+
+export const ExpireAndRefundModal: React.FC<ExpireAndRefundProps> = ({
+  active,
+  lock,
+  lockAddresses = [],
+  dismiss,
+}) => {
+  const [lockAddress] = lockAddresses
+  const walletService = useContext(WalletServiceContext)
+
+  const [refundAmount, setRefundAmount] = useState(0)
+  const [loading, setLoading] = useState(false)
+
+  const onAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setRefundAmount(parseFloat(e.target.value))
+  }
+
+  const onCloseCallback = () => {
+    if (typeof dismiss === 'function') dismiss()
+    setLoading(false)
+  }
+
+  const onExpireAndRefund = async () => {
+    const { keyholderAddress: keyOwner } = lock ?? {}
+    const amount = `${refundAmount}`
+    setLoading(true)
+
+    const params = {
+      lockAddress,
+      keyOwner,
+      amount,
+    }
+
+    try {
+      await walletService.expireAndRefundFor(params)
```

Comment:
You think toast.promise with progress might be good here?

Reply:
in this case not needed, when the "Expire and Refund" button is clicked. We still have the modal open until the promise fulfilled, and button will have a spinner to show the loading progress